bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -765,6 +910,36 @@ private void notifySensorStatusListeners(final FullSensor fullSensor, final Stri
         }
     }
 
+    private void notifyGroupStatusListeners(final FullGroup fullGroup, final String type) {
+        if (groupStatusListeners.isEmpty()) {
+            logger.debug("No group status listeners to notify of group change for group '{}'", fullGroup.getId());
+            return;
+        }
+
+        for (GroupStatusListener groupStatusListener : groupStatusListeners) {
+            try {
+                switch (type) {
+                    case STATE_ADDED:
+                        logger.debug("Sending groupAdded for group '{}'", fullGroup.getId());
+                        groupStatusListener.onGroupAdded(hueBridge, fullGroup);
+                        break;
+                    case STATE_GONE:
+                        groupStatusListener.onGroupGone(hueBridge, fullGroup);
+                        break;
+                    case STATE_CHANGED:
+                        logger.debug("Sending groupStateChanged for group '{}'", fullGroup.getId());
+                        groupStatusListener.onGroupStateChanged(hueBridge, fullGroup);
+                        break;
+                    default:
+                        throw new IllegalArgumentException(
+                                "Could not notify groupStatusListeners for unknown event type " + type);
+                }
+            } catch (Exception e) {
+                logger.error("An exception occurred while calling the Group Listeners", e);
```

Comment:
The error logging level should be reserved for issues catastrophic enough to threaten the operation of openHab itself. An error in a binding shouldn't qualify for this. Please read the [openHab coding guidelines](https://www.openhab.org/docs/developer/guidelines.html#f-logging) for expected use of logging in bindings.

My general rule for logging levels is:
* trace - used for normal expected execution paths
* debug - used for unexepected execution paths, but well within normal operation
* info - used for notable points in an execution path, like a milestone. (in openhab we try reserve this logging level for the core, so bindings should rarely ever call this.)
* warn - used for notable unexpected execution paths that a regular user (not just a developer) should be notified of. Warnings should be used to indicate that something not-normal occurred and user intervention is required to resolve. Warnings do not indicate a failure to operate merely an abnormal condition of operation that can still be handled by the binding. Failures in binding operation should be indicated by changing the thing status to offline.
* error - used to indicate catastrophic program failure. This should be used to indicate a catastrophic failure in openhab's ability to operate. A failure in a binding would never cause openhab as a whole to fail so a failure in a bindings should never log an error. Instead that failure should be indicated by changing the thing status.

Reply:
Ok, updated.